Implement the drawer component

Jay-flow opened the pull request:
### #30 I made a blueprint for the Drawer component.
The parts that have been made are as follows.

![ezgif com-gif-maker (1)](https://user-images.githubusercontent.com/29420674/128376020-5025e314-33df-457f-87ff-1611302311ae.gif)
---
This code has not yet been implemented for animation and is not implemented how it will behave when selecting a menu.
**Please let me know if you have any ideas or functions you want.**

hyochan commented:
Let's go to sleep. It's monday~

hyochan commented:
Also try to `rebase` your work here before further commits.

Jay-flow commented:
> Also try to `rebase` your work here before further commits.

I did as you said. I was able to practice on Rebase. Thank so much :)

Jay-flow commented:
**Current status**

https://user-images.githubusercontent.com/29420674/128903001-131fb0bf-e4bb-43e7-9f86-0c78f51c85b9.mp4

Is there any return value that we need other than URL when we select the menu?

hyochan commented on src/components/uis/Drawer/Menu.svelte:
```diff
@@ -0,0 +1,58 @@
+
+<div
+  class="sub-menu"
+  style="background-color: {isSelected ? '#F9F9F9' : '#FFFF'}"
```
Can we apply a theme color here?

Jay-flow replied:
> Can we apply a theme color here?

I checked. I changed it! :)

hyochan commented:
> **Current status**
> 
> Is there any return value that we need other than URL when we select the menu?
> 
>  2021-08-11.02-03-17.mp4

Looks enough for now currently.

Jay-flow commented:
### Added animation


https://user-images.githubusercontent.com/29420674/129204847-519224d9-d8c8-4179-9235-200e7113b720.mp4

hyochan reviewed (changes requested):
```
props: {
  communites,
}
```
Above props is only what we need I think.
Also, the menus are not variant, so you do need to pass them as a prop.

Jay-flow commented:
> 
> 
> ```
> props: {
>   communites,
> }
> ```
> 
> Above props is only what we need I think.
> Also, the menus are not variant, so you do need to pass them as a prop.

As you said, I changed the props of the drawer component :)

hyochan commented on src/components/uis/Drawer/Drawer.svelte:
```diff
@@ -0,0 +1,180 @@
+    selectMenu(menus[selectedCommunityId][0].url);
+  };
+
+  const requestMenuAPI = (communityId: string) => {
+    // eslint-disable-next-line no-console
+    console.debug(`Bind Menu via API Call (comminity id: ${communityId})`);
+
+    // Temporary Data
+    return [
+      {
+        name: 'Commits',
+        url: `https://github.com/wecount-dev/wecount/commits/${communityId}`,
+      },
+      {
+        name: 'Issues',
+        notificationCounts: 99,
+        url: `https://github.com/wecount-dev/wecount/issues/${communityId}`,
+      },
+    ];
+  };
```
right-side menus are currently not variant so I think we can make this to `const`.

Jay-flow replied:
> 
> 
> right-side menus are currently not variant so I think we can make this to `const`.

@hyochan 
As you said, I changed the menu data of the drawer component to constant. :)
I also deleted codes in the Temp.svelte to merge the code and I did the rebase.

---
**If you need a test, please use the code below.**
### Test plan `Temp.svelte`
```svelte
<script lang="ts">
  import Drawer from '../uis/Drawer/Drawer.svelte';

  const string = `This is <strong>Temp!</strong>`;

  const communites = [
    {
      id: '1',
      imageUrl:
        'https://cdn.icon-icons.com/icons2/2620/PNG/512/among_us_player_red_icon_156942.png',
    },
    {
      id: '2',
      imageUrl:
        'https://upload.wikimedia.org/wikipedia/commons/thumb/9/91/Octicons-mark-github.svg/2048px-Octicons-mark-github.svg.png',
    },
  ];

  const onSelectMenu = (path: string) => {
    // eslint-disable-next-line no-console
    console.log(`Selected Path: ${path}`);
  };
</script>

<div style="padding: 30px; background-color:#F2F5F6;">
  <p>{@html string}</p>

  <div>
    <Drawer communites={communites} onSelectMenu={onSelectMenu} />
  </div>
</div>

```

Jay-flow commented:
### Changes

**1. Added Dark Mode**

I thought white was better even in dark mode for the background color.

**default**
![default](https://user-images.githubusercontent.com/29420674/129926838-0a4981e6-f511-4a59-ba22-f62a0505fca6.png)

**dark**
![Screenshot from 2021-08-19 00-23-14](https://user-images.githubusercontent.com/29420674/129926921-7120896f-239c-4e6f-8eca-b6ea3297cbc7.png)


**2. Renamed translation variable: setting -> settings**

Jay-flow commented:
@hyochan @KoreanThinker @DevYuns 

### # Changes

**1. Added dark mode for drawer component**

default
![default](https://user-images.githubusercontent.com/29420674/129932369-c34841a3-db25-456d-9f99-c60d9a0b7968.png)

dark mode
![dark](https://user-images.githubusercontent.com/29420674/129932444-558b9f61-2c7c-44e0-b341-d9c57f976ecf.png)


**2. Renamed fake menu route & translation variables: setting -> settings**

DevYuns reviewed (approved):
LGTM!
